### ISSUE TITLE
discrete covariate split when use UpliftTreeClassifier

### ISSUE THREAD

**Issue by tianhong** [uber/causalml#132]
Thanks for the package, great contribution!

**Is your feature request related to a problem? Please describe.**
 I hava many covariates like age,gender,osid ,When I use UpliftTreeClassifier, i must use one-hot encoder just like pandas.get_dummies.  
When the model fit finished , it just use one level in a covariate.
`   X_names = ['age', 'gender', 'wnettype','absolute_rank',]
    for x in X_names:
        df[x] = df[x].astype('category')

    train_df = df.copy(True)
    uplift_model = UpliftTreeClassifier(max_depth=4,
                                        max_features=50,
                                        min_samples_leaf=500000,
                                        min_samples_treatment=250000,
                                        n_reg=100000,
                                        evaluationFunction='CTS',
                                        control_name='control')
    train_features_df = pd.get_dummies(train_df[X_names])

    uplift_model.fit(train_features_df.values,
                     treatment=train_df['treatment_group_key'].values,
                     y=train_df['click'].values)

    uplift_tree_string(uplift_model.fitted_uplift_tree, list(train_features_df.keys()))`

the output is
`absolute_rank_2 == 1?
yes -> {'treatment': xxx, 'control': xxx}
no  -> {'treatment': xxx, 'control': xxx}`

**Describe the solution you'd like**
How can I get a result just like this  picture
![image](https://user-images.githubusercontent.com/3253227/73608285-2d29a300-45fc-11ea-8e7a-dfa26a1abe2e.png)

**Comment by peterfoley** [comment]
CausalML uses the tree builder from sklearn, which is why it has the same numeric-columns-only limitations as sklearn's models like DecisionTreeClassifier. Dealing with categorial data in tree models is a big topic and aside from LightGBM, most tools don't allow for categorical variables, so people generally use [some other encoding scheme](https://towardsdatascience.com/know-about-categorical-encoding-even-new-ones-c266227b9cbd).

If you're interested in predictions and don't actually need the visualizations, I'd recommend creating multiple hashed or random numeric features for each level of your categorical variable (I don't like [sklearn's FeatureHasher](https://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.FeatureHasher.html), but it might work for you). A single hashed feature acts like label encoding, and a huge number of hashed features simulates searching over all possible partitions.

Visualization after those sorts of transformations can be a lot of work, though. Given a FeatureHasher that produces hash1, hash2, hash3, you'd recreate what values categories 1,2,...,10 will have for each hash, and at each branch, you'd back out which categorical values would have gone left or right. So the visualization is possible, but not particularly easy.

**Comment by tianhong** [comment]
> CausalML uses the tree builder from sklearn, which is why it has the same numeric-columns-only limitations as sklearn's models like DecisionTreeClassifier. Dealing with categorial data in tree models is a big topic and aside from LightGBM, most tools don't allow for categorical variables, so people generally use [some other encoding scheme](https://towardsdatascience.com/know-about-categorical-encoding-even-new-ones-c266227b9cbd).
> 
> If you're interested in predictions and don't actually need the visualizations, I'd recommend creating multiple hashed or random numeric features for each level of your categorical variable (I don't like [sklearn's FeatureHasher](https://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.FeatureHasher.html), but it might work for you). A single hashed feature acts like label encoding, and a huge number of hashed features simulates searching over all possible partitions.
> 
> Visualization after those sorts of transformations can be a lot of work, though. Given a FeatureHasher that produces hash1, hash2, hash3, you'd recreate what values categories 1,2,...,10 will have for each hash, and at each branch, you'd back out which categorical values would have gone left or right. So the visualization is possible, but not particularly easy.

Thank you very much！I will use Label Encoding!

Thanks for the package, great contribution!
